src/MultiFormats/MultiAddress.cs

```diff
@@ -0,0 +1,565 @@
+        /// </remarks>
+        private void Write(TextWriter stream)
+        {
+            foreach (var protocol in Protocols)
+            {
+                stream.Write('/');
+                stream.Write(protocol.Name);
+                protocol.WriteValue(stream);
+            }
+        }
+
+        /// <summary>
+        ///   Reads the binary representation of the the specified <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Stream"/> to read from.
+        /// </param>
+        /// <remarks>
+        ///   The binary representation is a sequence of <see cref="NetworkProtocol">network protocols</see>.
+        /// </remarks>
+        private void Read(Stream stream) { Read(new CodedInputStream(stream, true)); }
+
+        /// <summary>
+        ///   Reads the binary representation of the specified <see cref="Google.Protobuf.CodedInputStream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Google.Protobuf.CodedInputStream"/> to read from.
+        /// </param>
+        /// <remarks>
+        ///   The binary representation is a sequence of <see cref="NetworkProtocol">network protocols</see>.
+        /// </remarks>
+        private void Read(CodedInputStream stream)
+        {
+            Protocols.Clear();
+            do
+            {
+                var code = (uint) stream.ReadInt64();
+                if (!NetworkProtocol.Codes.TryGetValue(code, out var protocolType))
+                {
+                    throw new InvalidDataException($"The IPFS network protocol code '{code}' is unknown.");
+                }
+                
+                var p = (NetworkProtocol) Activator.CreateInstance(protocolType);
+                p.ReadValue(stream);
+                Protocols.Add(p);
+            } while (!stream.IsAtEnd);
+        }
+
+        /// <summary>
+        ///   Reads the string representation from the specified <see cref="TextReader"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="TextReader"/> to read from
+        /// </param>
+        /// <remarks>
+        ///   The string representation is a sequence of <see cref="NetworkProtocol">network protocols</see>.
+        /// </remarks>
+        private void Read(TextReader stream)
+        {
+            if (stream.Read() != '/')
+            {
+                throw new FormatException("An IPFS multiaddr must start with '/'.");
+            }
+
+            var name = new StringBuilder();
+            Protocols.Clear();
+            int c;
+            while (true)
+            {
+                name.Clear();
+                while (-1 != (c = stream.Read()) && c != '/')
+                {
+                    name.Append((char) c);
+                }
+
+                if (name.Length == 0)
+                {
+                    break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)